blaze/mahimahi/server/server.py

```diff
@@ -29,36 +29,43 @@ def prepend_javascript_snippet(input_string: str):
-function isAnyPartOfElementInViewport(el){var rect=el.getBoundingClientRect();var windowHeight=(window.innerHeight);var windowWidth=(window.innerWidth);console.log("alohomoradebug: window height is "+windowHeight)
-console.log("alohomoradebug: window width is "+windowWidth)
-var vertInView=(rect.top<windowHeight)&&((rect.top+rect.height)>=0);var horInView=(rect.left<windowWidth)&&((rect.left+rect.width)>=0);return(vertInView&&horInView)}
-function getCriticalRequests(){importantRequests=[]
-importantRequests=imagesInViewPort.map((url)=>(url));urlRequestors.forEach((k)=>{if(imagesInViewPort.includes(k.url)){importantRequests=importantRequests.concat(k.initiator)}})
-return importantRequests}
-function findAndPrintImagesInViewport(ele){ele.querySelectorAll('*').forEach(function(node){if(node.tagName=="IMG"){let url=null;console.log("alohomoradebug checking node "+JSON.stringify(node))
-if(isElementInViewport(node)){console.log("alohomoradebug: element is in viewport")
-if(typeof node.href!='undefined'){url=node.href}
-if(typeof node.src!='undefined'){url=node.src}
-if(url!=null){imagesInViewPort.push(url)}}else{console.log("alohomoradebug: element not in viewport")}}});let answer=getCriticalRequests()
-console.log("alohomoradebug: this is the answer")
-console.log(JSON.stringify({'alohomora_output':answer}))
-console.log("alohomoradebug: that was the the answer")}
-window.addEventListener('load',(event)=>{console.log("alohomoradebug: extracting critical elements")
-document.querySelectorAll("iframe").forEach((ele)=>{var iframeID=ele.id;iframeElement=document.getElementById(iframeID)
-if(isAnyPartOfElementInViewport(iframeElement)){var innerDoc=(iframeElement.contentDocument)?iframeElement.contentDocument:iframeElement.contentWindow.document;findAndPrintImagesInViewport(innerDoc)}});findAndPrintImagesInViewport(document)})
-    """
+
+    # in docker this should be /opt/blaze/blaze/mahimahi/server/injected-javascript
+    # in host, this should point to the folder in the machine
+    # eg. /home/murali/Code/blaze/blaze/mahimahi/server/injected-javascript
+    try:
+        dir_name = "/opt/blaze/blaze/mahimahi/server/injected-javascript"
+        stack_trace_dependency = soup.new_tag("script")
+        stack_trace_dependency["type"] = "application/javascript"
+        with open(os.path.join(dir_name, "1-stacktrace.js")) as f:
+            stack_trace_dependency.string = f.read()
+        interceptor = soup.new_tag("script")
+        interceptor["type"] = "application/javascript"
+        with open(os.path.join(dir_name, "2-interceptor.js")) as f:
+            interceptor.string = f.read()
+        critical_catcher = soup.new_tag("script")
+        critical_catcher["type"] = "application/javascript"
+        with open(os.path.join(dir_name, "3-find-in-viewport.js")) as f:
+            critical_catcher.string = f.read()
+    except FileNotFoundError:
```

Comment:
remove this try/except with the alternate path -- can add this in when testing locally